### PR TITLE
777 fix filters

### LINKED ISSUE #777: [Elasticsearch-connector] setFilter not working
**Describe the bug**
The setFilter action is not building the right Elasticsearch queries.

**To Reproduce**

Trigger setFilter for a long field

```
    setFilter("precio", {
      name: "precio",
      from: rangePrices[0],
      to: rangePrices[1],
    });
```

**Expected behavior**
A "range" query should trigger because the filter value is an object, but a term query is triggered instead. Also, the term query throws an error because "term" does not support "name" which is required in setFilter.

**Screenshots**

**Which backends and packages are you using:**
Backend: Elasticsearch
Packages: search-ui-elasticsearchsearch-connector,

### PR DESCRIPTION
**Pull request by joemcelroy**
fixes #777

### CODE REVIEW

File: packages/search-ui-elasticsearch-connector/package.json [hunk 1]
```diff
@@ -41,6 +41,6 @@
   "dependencies": {
     "@elastic/elasticsearch": "^8.2.1",
     "@elastic/search-ui": "1.15.0",
-    "@searchkit/sdk": "^3.0.0-canary.57"
+    "@searchkit/sdk": "^3.0.0-canary.58"
```
Comment (JasonStoltz) [inline packages/search-ui-elasticsearch-connector/package.json:44]: Why do you use the canary versions in SearchKit?

**Review by JasonStoltz — COMMENTED** [review]
I see:
1) Code
2) Tests
3) Updated example

All looks well to me. Is there anything we need to note in our docs about this?

File: packages/search-ui-elasticsearch-connector/package.json (hunk 1 above)
Comment (joemcelroy) [reply packages/search-ui-elasticsearch-connector/package.json:44]: searchkit v3 not yet pass pre-release yet so still on canaries right now.

**Review by JasonStoltz — DISMISSED** [review]
Hey Joe, I thought I added my review here but it looks like I did not. Everything looks good I'm just wondering if you need to update anything in docs?

**Comment by joemcelroy** [comment]
> Hey Joe, I thought I added my review here but it looks like I did not. Everything looks good I'm just wondering if you need to update anything in docs?

Yeh its a good shout. I've added it to the connector api docs + another known issue. Hope these changes are ok now.

**Review by JasonStoltz — APPROVED** [review]

**Pull request merged**